### PR TITLE
Post "Linux Mint on a Razer Blade"

### PR DESCRIPTION
**Pull request by likethemammal**

### CODE REVIEW

File: _posts/2016-04-09-linux-mint-razer-blade.markdown [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+Load up 4.2.0-18 and then restart the machine. Mint should automatically boot into the latest version.
+
+This should fix the touchpad and suspend issues. The internal webcam issue still haunts me. I've worked around it by booting into Windows (works fine there) or using an external one. There are [several](https://insider.razerzone.com/index.php?threads/linux-on-a-new-blade.4115/page-4) [threads](https://bugs.launchpad.net/ubuntu/+source/cheese/+bug/1295247) that have mentioned this is as a common issue.
+
+###Chapter 4: Fixing software issues
+As I'll discuss in the last chapter, battery length a bit disappointing at first. With Cinnamon, by default I was barely getting 1.5 hours of battery life with general usage (internet browsing, videos, code editor)! Maybe 2.5 using Windows. At first this felt like a deal breaker, until I fixed the issues actually causing the power drain. Now I can get around 3.5 - 4. Which, in my opinion, is decent for a laptop gaming rig.
```
Comment (mediocregopher) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: "battery length *was* a bit disappointing"

**Comment by mediocregopher** [comment]
1 comment, looks good though! Does the amazon link have your referral code in it?

**Comment by likethemammal** [comment]
Yea, its an Amazon Associates special link. Thought that, since I was talkin about it, ppl might be considering getting one. Might as well get me a slice :pizza: :moneybag: :pizza:

File: _posts/2016-04-09-linux-mint-razer-blade.markdown [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+Plug the Razer Blade into a power source. Follow the directions in the user guide to boot from USB and start the install process.
+
+If you end up on the partition menu to customize the size of the install, make sure to add a few GB partition as "swap area". Among other things, this is important for resuming from a suspend. Learned this the hard way.
+
+Follow the rest of the directions to install completely and remove the USB drive (it'll prompt you).
+
+After install, boot up the machine, sign in, and install any updates from the Update Manager.
+
+###Chapter 3: Fixing hardware issues
+There are several parts of the hardware that just aren't hooked up right with the base install. I ran into issues with suspend black-screening indefinitely, two-finger scroll not working and other touchpad issues, and the internal webcam not working.
```
Comment (kinghrothgar) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: Nit pick, but I believe the more common term is "blank-screening"

File: _posts/2016-04-09-linux-mint-razer-blade.markdown [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+###Chapter 4: Fixing software issues
+As I'll discuss in the last chapter, battery length a bit disappointing at first. With Cinnamon, by default I was barely getting 1.5 hours of battery life with general usage (internet browsing, videos, code editor)! Maybe 2.5 using Windows. At first this felt like a deal breaker, until I fixed the issues actually causing the power drain. Now I can get around 3.5 - 4. Which, in my opinion, is decent for a laptop gaming rig.
+
+**Switching to Nvidia drivers**
+
+At some point, it may be explicitly beneficial for you to switch to the Nvidia video card drivers over the default open-source XOrg drivers. Switching will also provide more fine grain control over the video card and provides the ability to switch between Performance settings and Power Saving settings.
```
Comment (kinghrothgar) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: I believe it's either X.Org (how they actually stylize it) or everywhere else it's just Xorg

File: _posts/2016-04-09-linux-mint-razer-blade.markdown [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+Theres loads of stuff that can be done to reduce the wear on a SSD. Several things on [this list](https://sites.google.com/site/easylinuxtipsproject/ssd) can, and should, only be done before install. Its something to consider.
+
+**Get bootable USB drive**
+
+Get the download for Mint from the [downloads page](https://www.linuxmint.com/download.php), I **highly recommend** you choose the Mate version over Cinnamon. This will come into play later. Although, if you choose Cinnamon and run into problems, there're ways to switch after the fact.
+
+Open up the Linux Mint [User Guide](https://www.linuxmint.com/documentation.php) and follow the instructions to verify your MD5. You'll also want this open for details on most of the initial setup process. This post only covers steps beyond the guide.
+
+Use [this tutorial](https://community.linuxmint.com/tutorial/view/744) to mount the downloaded `.iso` to a USB drive thats at least 16GB.
```
Comment (kinghrothgar) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: Only needs a 2GB

File: _posts/2016-04-09-linux-mint-razer-blade.markdown (hunk 2 above)
Comment (likethemammal) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: Think I'm going to leave this this way. I wouldn't have known to look up "blank screening" when I was having the issue

File: _posts/2016-04-09-linux-mint-razer-blade.markdown (hunk 3 above)
Comment (likethemammal) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: thanks for the catch

**Comment by likethemammal** [comment]
i didn't close this, idk what happened

**Comment by likethemammal** [comment]
trying to reopen

File: _posts/2016-04-09-linux-mint-razer-blade.markdown [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+A few months back Razer finally put that model up <a  href="http://www.amazon.com/gp/product/B01AGINE7Q/ref=as_li_tl?ie=UTF8&camp=1789&creative=9325&creativeASIN=B01AGINE7Q&linkCode=as2&tag=likethemammal-20&linkId=VLVDTKUEZSESNTID">on Amazon</a><img src="http://ir-na.amazon-adsystem.com/e/ir?t=likethemammal-20&l=as2&o=1&a=B01AGINE7Q" width="1" height="1" border="0" alt="" style="border:none !important; margin:0px !important;" />, at a discount. With the power of a credit card, I finally had the mini gaming workstation of my dreams in hand!
+
+I've done 3 or 4 full builds of Linux before, all with Mint. One on a Chromebook, but mostly on machines with slightly dated hardware. This build has by far been the hardest, and I learned the hard way that Linux isn't as stable on "new" hardware.
+
+###Chapter 1: Pre-install
+**Expand SSD storage (optional)**
+
+The SSD in the Full HD edition is about 250 GB. If you're dual-booting, like I did, this doesn't leave much space to split. If you're going to replace the ssd with a larger one, now's the time to do it. Heres a video on how:
```
Comment (tash1207) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: *Here's

File: _posts/2016-04-09-linux-mint-razer-blade.markdown [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/xjM-ByOE0fQ" frameborder="0" allowfullscreen></iframe>
+
+**Prepare your SSD (optional)**
+
+Theres loads of stuff that can be done to reduce the wear on a SSD. Several things on [this list](https://sites.google.com/site/easylinuxtipsproject/ssd) can, and should, only be done before install. Its something to consider.
```
Comment (tash1207) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: *There's and *It's

File: _posts/2016-04-09-linux-mint-razer-blade.markdown [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+**Get bootable USB drive**
+
+Get the download for Mint from the [downloads page](https://www.linuxmint.com/download.php), I **highly recommend** you choose the Mate version over Cinnamon. This will come into play later. Although, if you choose Cinnamon and run into problems, there're ways to switch after the fact.
+
+Open up the Linux Mint [User Guide](https://www.linuxmint.com/documentation.php) and follow the instructions to verify your MD5. You'll also want this open for details on most of the initial setup process. This post only covers steps beyond the guide.
+
+Use [this tutorial](https://community.linuxmint.com/tutorial/view/744) to mount the downloaded `.iso` to a USB drive thats at least 2GB.
```
Comment (tash1207) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: *that's

File: _posts/2016-04-09-linux-mint-razer-blade.markdown [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+If you end up on the partition menu to customize the size of the install, make sure to add a few GB partition as "swap area". Among other things, this is important for resuming from a suspend. Learned this the hard way.
+
+Follow the rest of the directions to install completely and remove the USB drive (it'll prompt you).
+
+After install, boot up the machine, sign in, and install any updates from the Update Manager.
+
+###Chapter 3: Fixing hardware issues
+There are several parts of the hardware that just aren't hooked up right with the base install. I ran into issues with suspend black-screening indefinitely, two-finger scroll not working and other touchpad issues, and the internal webcam not working.
+
+The only concrete and consistent fix I've found is to change the version of the Linux kernel thats being used. This is a critical part of the system, so use caution, but I was able to fix most of my hardware issues this way.
```
Comment (tash1207) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: *that's

File: _posts/2016-04-09-linux-mint-razer-blade.markdown [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+
+**Changing the kernel**
+
+Using the same Update Manager, navigate to View > Linux Kernels. This menu shows the full list of kernels available to switch to. As of this writing, the recommended and default kernel for Mint is **3.19.0-32**. I tried several different versions, some fixing some issues and creating others. The version I found most stable is **4.2.0-18**, and that's what I landed on.
+
+Load up 4.2.0-18 and then restart the machine. Mint should automatically boot into the latest version.
+
+This should fix the touchpad and suspend issues. The internal webcam issue still haunts me. I've worked around it by booting into Windows (works fine there) or using an external one. There are [several](https://insider.razerzone.com/index.php?threads/linux-on-a-new-blade.4115/page-4) [threads](https://bugs.launchpad.net/ubuntu/+source/cheese/+bug/1295247) that have mentioned this is as a common issue.
```
Comment (tash1207) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: that have mentioned this is a common issue

File: _posts/2016-04-09-linux-mint-razer-blade.markdown [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+
+###Chapter 4: Fixing software issues
+As I'll discuss in the last chapter, battery length was a bit disappointing at first. With Cinnamon, by default I was barely getting 1.5 hours of battery life with general usage (internet browsing, videos, code editor)! Maybe 2.5 using Windows. At first this felt like a deal breaker, until I fixed the issues actually causing the power drain. Now I can get around 3.5 - 4. Which, in my opinion, is decent for a laptop gaming rig.
+
+**Switching to Nvidia drivers**
+
+At some point, it may be explicitly beneficial for you to switch to the Nvidia video card drivers over the default open-source X.Org drivers. Switching will also provide more fine grain control over the video card and provides the ability to switch between Performance settings and Power Saving settings.
+
+Mint comes installed with Driver Manger. Open that up. There should only be the one set of video driver options. The OS recommends switching to the `nvidia-352` drivers but I went with `nvidia-352-updates` just to get the latest fixes.
```
Comment (tash1207) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: *Manager

File: _posts/2016-04-09-linux-mint-razer-blade.markdown [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+
+Apply the changes and restart when it prompts.
+
+**Switching from Cinnamon**
+
+I made the mistake of installing the Cinnamon version of Mint, which led to some problems. Cinnamon, just like Mate, is a display manager for the OS. Its a bit cleaner and chicer than Mate, but I ran into **huge** performance issues with Cinnamon. This screenshot below depicts the issue.
```
Comment (tash1207) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: *It's a bit

File: _posts/2016-04-09-linux-mint-razer-blade.markdown [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+###Chapter 5: Power savers
+The rest are just tips for getting the most out of the battery.
+
+**Set the autostart brightness**
+
+By default the brightness of the screen is all the way up. I rarely want it that high, so I set the autostart brightness to about 40%. That way I don't have to remember to lower it everytime. [This answer](http://askubuntu.com/a/151665) explains how-to (the comments explain how to find your brightness levels).
+
+**Consider disabling bluetooth**
+
+Keeping bluetooth running all the time chews up battery life, just like it does on your phone. Theres [several ways](https://community.linuxmint.com/tutorial/view/1102) to disable bluetooth in Linux Mint.
```
Comment (tash1207) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: There are several ways

File: _posts/2016-04-09-linux-mint-razer-blade.markdown [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+
+###Epilogue
+Thats it. After months of tweaking, and 3 re-builds of the same machine I've come out happy with my Razer Blade. As I mentioned, my only issue left is the internal webcam. Hopefully, as the hardware becomes more supported, the kernels will fix this issue.
```
Comment (tash1207) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: *That's

File: _posts/2016-04-09-linux-mint-razer-blade.markdown [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/xjM-ByOE0fQ" frameborder="0" allowfullscreen></iframe>
+
+**Prepare your SSD (optional)**
+
+Theres loads of stuff that can be done to reduce the wear on a SSD. Several things on [this list](https://sites.google.com/site/easylinuxtipsproject/ssd) can, and should, only be done before install. Its something to consider.
+
+**Get bootable USB drive**
+
+Get the download for Mint from the [downloads page](https://www.linuxmint.com/download.php), I **highly recommend** you choose the Mate version over Cinnamon. This will come into play later. Although, if you choose Cinnamon and run into problems, there're ways to switch after the fact.
```
Comment (akenn) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: Thererererere

File: _posts/2016-04-09-linux-mint-razer-blade.markdown [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+layout: post
+title: Running Linux Mint on a Razer Blade
+tags: linux mint razer-blade setup
+username: likethemammal
+---
+
+![Linux Mint logo](https://upload.wikimedia.org/wikipedia/commons/5/5c/Linux_Mint_Official_Logo.svg)
+
+**tl;dr** The <a  href="http://www.amazon.com/gp/product/B01AGINE7Q/ref=as_li_tl?ie=UTF8&camp=1789&creative=9325&creativeASIN=B01AGINE7Q&linkCode=as2&tag=likethemammal-20&linkId=VLVDTKUEZSESNTID">Razer Blade 2015</a><img src="http://ir-na.amazon-adsystem.com/e/ir?t=likethemammal-20&l=as2&o=1&a=B01AGINE7Q" width="1" height="1" border="0" alt="" style="border:none !important; margin:0px !important;" /> is a great portable gaming workstation. Linux Mint runs beautifully with some tweaking, that I've boiled down after months of usage. The only problem still unresolved is an issue with the internal webcam.
+
```
Comment (akenn) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: "that I've boiled down after months of usage" reads a little weird. I'd kill it.

File: _posts/2016-04-09-linux-mint-razer-blade.markdown (hunk 5 above)
Comment (akenn) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: If you're like me and want to dual-boot, this doesn't leave much space for each partition.

File: _posts/2016-04-09-linux-mint-razer-blade.markdown [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+**Prepare your SSD (optional)**
+
+Theres loads of stuff that can be done to reduce the wear on a SSD. Several things on [this list](https://sites.google.com/site/easylinuxtipsproject/ssd) can, and should, only be done before install. Its something to consider.
+
+**Get bootable USB drive**
+
+Get the download for Mint from the [downloads page](https://www.linuxmint.com/download.php), I **highly recommend** you choose the Mate version over Cinnamon. This will come into play later. Although, if you choose Cinnamon and run into problems, there're ways to switch after the fact.
+
+Open up the Linux Mint [User Guide](https://www.linuxmint.com/documentation.php) and follow the instructions to verify your MD5. You'll also want this open for details on most of the initial setup process. This post only covers steps beyond the guide.
```
Comment (akenn) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: The official guide is a pdf? wtf?

File: _posts/2016-04-09-linux-mint-razer-blade.markdown [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+
+Use [this tutorial](https://community.linuxmint.com/tutorial/view/744) to mount the downloaded `.iso` to a USB drive thats at least 2GB.
+
+###Chapter 2: Install
+Plug the Razer Blade into a power source. Follow the directions in the user guide to boot from USB and start the install process.
+
+If you end up on the partition menu to customize the size of the install, make sure to add a few GB partition as "swap area". Among other things, this is important for resuming from a suspend. Learned this the hard way.
```
Comment (akenn) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: What's a few GB? 2 GB? 3 GB? I'd also include the value you used.

File: _posts/2016-04-09-linux-mint-razer-blade.markdown (hunk 8 above)
Comment (akenn) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: I've found is to update the version of the Linux kernel to a more recent version.

File: _posts/2016-04-09-linux-mint-razer-blade.markdown [hunk 18]
```diff
@@ -0,0 +1,97 @@
...
+
+Load up 4.2.0-18 and then restart the machine. Mint should automatically boot into the latest version.
+
+This should fix the touchpad and suspend issues. The internal webcam issue still haunts me. I've worked around it by booting into Windows (works fine there) or using an external one. There are [several](https://insider.razerzone.com/index.php?threads/linux-on-a-new-blade.4115/page-4) [threads](https://bugs.launchpad.net/ubuntu/+source/cheese/+bug/1295247) that have mentioned this is as a common issue.
+
+###Chapter 4: Fixing software issues
+As I'll discuss in the last chapter, battery length was a bit disappointing at first. With Cinnamon, by default I was barely getting 1.5 hours of battery life with general usage (internet browsing, videos, code editor)! Maybe 2.5 using Windows. At first this felt like a deal breaker, until I fixed the issues actually causing the power drain. Now I can get around 3.5 - 4. Which, in my opinion, is decent for a laptop gaming rig.
```
Comment (akenn) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: Is this paragraph needed if you cover it more in-depth later on?

File: _posts/2016-04-09-linux-mint-razer-blade.markdown [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+
+**Switching to Nvidia drivers**
+
+At some point, it may be explicitly beneficial for you to switch to the Nvidia video card drivers over the default open-source X.Org drivers. Switching will also provide more fine grain control over the video card and provides the ability to switch between Performance settings and Power Saving settings.
```
Comment (akenn) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: Elaborate on the differences between Performance settings and Power Saving settings. How much longer is the batter life when you chose Power Saving mode? Are games slow when you're in Power Saving mode? How slow?

Also what you mean by that "explicitly beneficial". Beneficial for you choose a specific version of a video card driver?

File: _posts/2016-04-09-linux-mint-razer-blade.markdown (hunk 14 above)
Comment (akenn) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: What will come in to play later? What are Mate and Cinnamon? Mention that Mint gives you the option to choose your window manager, and that you recommend Mate.

File: _posts/2016-04-09-linux-mint-razer-blade.markdown [hunk 20]
```diff
@@ -0,0 +1,97 @@
...
+
+Apply the changes and restart when it prompts.
+
+**Switching from Cinnamon**
+
+I made the mistake of installing the Cinnamon version of Mint, which led to some problems. Cinnamon, just like Mate, is a display manager for the OS. Its a bit cleaner and chicer than Mate, but I ran into **huge** performance issues with Cinnamon. This screenshot below depicts the issue.
+
+![System Monitor, high cpu](http://i.imgur.com/aDXBPoQ.png)
+
+CPU usage would start oscillating randomly. I noticed because, without any high-demand programs running, scrolling and typing would start lagging. Every time it happened I opened up the System Monitor and found this same pattern. The process that was causing this CPU fluctuation was Cinnamon. I tried a myriad of commands, tweaks, and tricks to try to fix the issue, but was never successful. I ended up switching to Mate. Resolving the issues completely.
```
Comment (akenn) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: I ended up switching to Mate, which resolved these issues completely.

File: _posts/2016-04-09-linux-mint-razer-blade.markdown (hunk 10 above)
Comment (MarcoPolo) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: * Mr. Manager

Comment (tash1207) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: We just say Manager

File: _posts/2016-04-09-linux-mint-razer-blade.markdown (hunk 19 above)
Comment (likethemammal) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: I don't know the differences between them really. I know they exist but I haven't done much testing with them. I just wanted to mention that they were around.

Explicitly beneficial as in a tutorial or a forum might suggest switching drivers to fix a specific issue someone might be having.

File: _posts/2016-04-09-linux-mint-razer-blade.markdown (hunk 17 above)
Comment (likethemammal) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: I'm going to leave this at "a few" cus I don't wanna be specific. Some places say 10Gbs, some say you don't need more than 2 or 3. This post isn't really about swap space, I just know if you don't have any suspend will get fucked up.

File: _posts/2016-04-09-linux-mint-razer-blade.markdown (hunk 16 above)
Comment (likethemammal) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: Yea, idk whats up with the pdf. I guess its easy to include on a disk then

File: _posts/2016-04-09-linux-mint-razer-blade.markdown (hunk 15 above)
Comment (likethemammal) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: I want to emphasize that it took me time and effort to get this list of tweaks together. It took months of trying commands, rebuilding from scratch, and frustration.

How would you suggest I break up the sentence and also get this across?

File: _posts/2016-04-09-linux-mint-razer-blade.markdown (hunk 9 above)
Comment (likethemammal) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: went with "as a common issue" cus noone in the forum actually specific said it was common. Enough comments in the thread mentioned to suggest that it was common.

File: _posts/2016-04-09-linux-mint-razer-blade.markdown (hunk 8 above)
Comment (likethemammal) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: Didn't use the word "update" cus it suggests that its a change to a stable version. Its a change to a more recent version, but I won't call it an "update"

File: _posts/2016-04-09-linux-mint-razer-blade.markdown (hunk 5 above)
Comment (likethemammal) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: Don't wanna be specific about the number of gigs cus the os's and rando partitions take up space here or there. Its probably closer to 100-110 each.

Also, the partitions don't have to be split down the middle. Just wanna suggest that 250 is much to split between two os's

File: _posts/2016-04-09-linux-mint-razer-blade.markdown (hunk 18 above)
Comment (likethemammal) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: Changed the wording of the first sentence

File: _posts/2016-04-09-linux-mint-razer-blade.markdown (hunk 15 above)
Comment (akenn) [inline _posts/2016-04-09-linux-mint-razer-blade.markdown]: hmm this is the best I could come up with. Feel free to ignore it if you like what you have more.

After months of researching and ironing out the kinks, I've found that Linux Mint runs beautifully with some tweaking. The only problem still unresolved is an issue with the internal webcam.

**Comment by akenn** [comment]
LGTM! 🚀

**Pull request merged**